src/shared/generator.c

```diff
@@ -708,6 +708,108 @@ int generator_hook_up_pcrfs(
         return generator_add_symlink_full(dir, where_unit, "wants", pcrfs_unit_path, instance);
 }
 
+int generator_hook_up_quotaservice(
+                const char *dir,
+                const char *where,
+                const char *target) {
+
+        _cleanup_free_ char *q_check_unit = NULL, *q_on_unit = NULL, *escaped = NULL, *where_unit = NULL, *q_check_unit_file = NULL, *q_on_unit_file =NULL;
+        _cleanup_fclose_ FILE *f = NULL;
+        int r;
+
+        assert(dir);
+        assert(where);
+
+        escaped = cescape(where);
+        if (!escaped)
+                return log_oom();
+
+        r = unit_name_from_path_instance("systemd-quotacheck", where, ".service", &q_check_unit);
+        if (r < 0)
+                return log_error_errno(r, "Failed to make unit instance name from path \"%s\": %m",
+                                       where);
+
+        r = unit_name_from_path(where, ".mount", &where_unit);
+        if (r < 0)
+                return log_error_errno(r, "Failed to make unit name from path \"%s\": %m",
+                                       where);
+
+        q_check_unit_file = path_join(dir, q_check_unit);
+        if (!q_check_unit_file)
+                return log_oom();
+
+        log_debug("Creating %s", q_check_unit_file);
+
+        f = fopen(q_check_unit_file, "wxe");
+        if (!f)
+                return log_error_errno(errno, "Failed to create unit file %s: %m",
+                                       q_check_unit_file);
+
+        fprintf(f,
+                "# Automatically generated by %s\n\n"
+                "[Unit]\n"
+                "Description=File System Quota Check\n"
+                "Documentation=man:systemd-quotacheck.service(8)\n"
+                "DefaultDependencies=no\n"
+                "After=systemd-remount-fs.service\n"
+                "Before=remote-fs.target shutdown.target\n",
```

Comment:
I thought that the problem that we wanted to fix initially was that depending on the nature of the mount unit (local vs remote) we wanted to order the quota service before either `local-fs.target` or `remote-fs.target`.

But it seems that it's unconditionally ordered before `remote-fs.target` here.